smartsim/_core/generation/generator.py

```diff
@@ -236,67 +314,51 @@ def _symlink_files(files: t.Union[EntityFiles, None], dest: pathlib.Path) -> Non
 
-        :param app: The Application attached to the Job
-        :param dest: Path to the Jobs run directory
+        This method processes tagged files by reading their configurations,
+        serializing the provided parameters, and generating commands to
+        write these configurations to the destination directory.
+
+        :param files: An EntityFiles object containing the paths to configure, or None.
+        :param params: A dictionary of params
+        :param dest: The destination path to the Job's run directory.
+        :return: A CommandList containing the configuration commands, or None if no files are provided.
         """
-        # Return if no files are attached
         if files is None:
-            return
+            return None
+        cmd_list = CommandList()
         if files.tagged:
-            to_write = []
-
-            def _build_tagged_files(tagged: TaggedFilesHierarchy) -> None:
-                """Using a TaggedFileHierarchy, reproduce the tagged file
-                directory structure
-
-                :param tagged: a TaggedFileHierarchy to be built as a
-                               directory structure
-                """
-                for file in tagged.files:
-                    dst_path = path.join(dest, tagged.base, path.basename(file))
-                    shutil.copyfile(file, dst_path)
-                    to_write.append(dst_path)
-
-                for tagged_dir in tagged.dirs:
-                    mkdir(path.join(dest, tagged.base, path.basename(tagged_dir.base)))
-                    _build_tagged_files(tagged_dir)
-
-            if files.tagged_hierarchy:
-                _build_tagged_files(files.tagged_hierarchy)
-
-            # Pickle the dictionary
+            tag_delimiter = ";"
             pickled_dict = pickle.dumps(params)
-            # Default tag delimiter
-            tag = ";"
-            # Encode the pickled dictionary with Base64
             encoded_dict = base64.b64encode(pickled_dict).decode("ascii")
-            for dest_path in to_write:
-                subprocess.run(
-                    args=[
+            for path in files.tagged:
+                cmd = Command(
+                    [
                         sys.executable,
                         "-m",
                         "smartsim._core.entrypoints.file_operations",
                         "configure",
-                        dest_path,
-                        dest_path,
-                        tag,
+                        path,
+                        str(dest),
+                        tag_delimiter,
                         encoded_dict,
                     ]
                 )
+                cmd_list.commands.append(cmd)
+        return cmd_list
 
-            # TODO address in ticket 723
-            # self._log_params(entity, files_to_params)
+        # TODO address in ticket 723
```

Comment:
Can we add this code down here to ticket 723 and delete?

Reply:
yes!